### PR TITLE
Update wasm.md to include some HTTP instructions

### PR DESCRIPTION
**Pull request by brendandburns**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)

In addition, please fill out the following to help reviewers understand this pull request:

## Description
Add docs for recent WASM updates.

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 1]
```diff
@@ -41,6 +41,15 @@ Dapr uses [wazero](https://wazero.io) to run these binaries, because it has no
 dependencies. This allows use of WebAssembly with no installation process
 except Dapr itself.
 
+The WASM output binding supports making HTTP client calls using the [wasi-http](https://github.com/WebAssembly/wasi-http) specification.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
The Wasm output binding supports making HTTP client calls using the [wasi-http](https://github.com/WebAssembly/wasi-http) specification.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 2]
```diff
@@ -41,6 +41,15 @@ Dapr uses [wazero](https://wazero.io) to run these binaries, because it has no
...
+You can find example code for making HTTP calls in a variety of languages here:
+* [Golang](https://github.com/dev-wasm/dev-wasm-go/tree/main/http)
+* [C](https://github.com/dev-wasm/dev-wasm-c/tree/main/http)
+* [dotnet](https://github.com/dev-wasm/dev-wasm-dotnet/tree/main/http)
+* [TypeScript](https://github.com/dev-wasm/dev-wasm-ts/tree/main/http)
+
+Note that if you just want to make an HTTP call, it is far simpler to use the [HTTP Binding](https://docs.dapr.io/reference/components-reference/supported-bindings/http/) but if you need to add your own logic, for example filtering or calling to different api endpoints, then it makes sense to use WebAssembly.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
Note that if you just want to make an HTTP call, it is simpler to use the [service invocation API]({{< ref howto-invoke-non-dapr-endpoints.md >}}). However if you need to add your own logic, for example filtering or calling to multiple API endpoints, then consider using WebAssembly.
```

**Review by msfussell — COMMENTED** [review]
@brendandburns - Given this binding change is in the v1.12 release, can you target this PR as the v1.12 docs branch? Thx

**Comment by brendandburns** [comment]
hrmph, switching bases messed up this PR. I will re-compute it...

**Comment by brendandburns** [comment]
Ok, fixed.

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 3]
```diff
@@ -41,6 +41,15 @@ Dapr uses [wazero](https://wazero.io) to run these binaries, because it has no
 dependencies. This allows use of WebAssembly with no installation process
 except Dapr itself.
 
+The Wasm output binding supports making HTTP client calls using the [wasi-http](https://github.com/WebAssembly/wasi-http) specification.
+You can find example code for making HTTP calls in a variety of languages here:
+* [Golang](https://github.com/dev-wasm/dev-wasm-go/tree/main/http)
+* [C](https://github.com/dev-wasm/dev-wasm-c/tree/main/http)
+* [dotnet](https://github.com/dev-wasm/dev-wasm-dotnet/tree/main/http)
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
* [.NET](https://github.com/dev-wasm/dev-wasm-dotnet/tree/main/http)
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md [hunk 4]
```diff
@@ -41,6 +41,15 @@ Dapr uses [wazero](https://wazero.io) to run these binaries, because it has no
...
+* [TypeScript](https://github.com/dev-wasm/dev-wasm-ts/tree/main/http)
+
+Note that if you just want to make an HTTP call, it is simpler to use the [service invocation API]({{< ref howto-invoke-non-dapr-endpoints.md >}}). However if you need to add your own logic, for example filtering or calling to multiple API endpoints, then consider using WebAssembly.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md]:
```suggestion
{{% alert title="Note" color="primary" %}}
If you just want to make an HTTP call, it is simpler to use the [service invocation API]({{< ref howto-invoke-non-dapr-endpoints.md >}}). However, if you need to add your own logic - for example, filtering or calling to multiple API endpoints - consider using Wasm.
{{% /alert %}}
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]

**Comment by brendandburns** [comment]
Thanks for the suggestions, merged.

**Review by hhunter-ms — APPROVED** [review]

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**